src/main/java/org/broadinstitute/hellbender/engine/filters/CountedReadFilter.java

```diff
@@ -0,0 +1,165 @@
+        Objects.requireNonNull(other);
+        return new CountedOrReadFilter(this, other);
+    }
+
+    /**
+     * Specialization of negate so that the resulting object is still a CountedReadFilter
+     */
+    @Override
+    public CountedReadFilter negate() {
+        return new CountedReadFilter("(Not '" + getName() + "')",(t) -> !test(t));
+    }
+
+    @Override
+    public boolean test(final GATKRead read) {
+        final boolean accept = delegateFilter.test(read);
+        if (!accept) {
+            filteredCount++;
+        }
+        return accept;
+    }
+
+    /**
+     * Private class for counted binary operator (and/or) filters; these keep track of how many reads are filtered at
+     * each level of filter nesting.
+     */
+    private abstract class CountedBinopReadFilter extends CountedReadFilter {
+
+        static final long serialVersionUID = 1L;
+
+        final CountedReadFilter lhs;
+        final CountedReadFilter rhs;
+
+        public CountedBinopReadFilter(final String name, final CountedReadFilter lhs, final CountedReadFilter rhs) {
+            super(name);
+            this.lhs = lhs;
+            this.rhs = rhs;
+        }
+
+        @Override
+        protected String getSummaryLineForLevel(final int indentLevel) {
+            final String indent = getIndentString(indentLevel);
+            return filteredCount == 0 ?
+                    "" :
+                    indent + Long.toString(filteredCount) + " read(s) filtered by: " + getName() + "\n"
+                            + (lhs.getFilteredCount() > 0 ? indent + lhs.getSummaryLineForLevel(indentLevel + 1) : "")
+                            + (rhs.getFilteredCount() > 0 ? indent + rhs.getSummaryLineForLevel(indentLevel + 1) : "");
+        }
+    }
+
+    /**
+     * Private class for Counted AND filters
+     */
+    private class CountedAndReadFilter extends CountedBinopReadFilter {
+
+        static final long serialVersionUID = 1L;
+
+        private CountedAndReadFilter(final CountedReadFilter lhs, final CountedReadFilter rhs) {
+            super("(" + lhs.getName() + " AND " + rhs.getName() + ")", lhs, rhs);
+        }
+
+        @Override
+        public boolean test(final GATKRead read) {
+            final boolean accept = lhs.test(read) && rhs.test(read);
+            if (!accept) {
+                filteredCount++;
+            }
+            return accept;
+        }
+    }
+
+    /**
+     * Private class for Counted OR filters
+     */
+    private class CountedOrReadFilter extends CountedBinopReadFilter {
+
+        static final long serialVersionUID = 1L;
+
+        private CountedOrReadFilter(final CountedReadFilter lhs, final CountedReadFilter rhs) {
+            super("(" + lhs.getName() + " OR " + rhs.getName() + ")", lhs, rhs);
+        }
+
+        @Override
+        public boolean test(final GATKRead read) {
+            final  boolean accept = lhs.test(read) || rhs.test(read);
```

Comment:
Same comment here -- use non-short-circuiting or for the sake of producing accurate counts.